test/built-ins/Set/prototype/union/add-not-called.js

```diff
@@ -0,0 +1,31 @@
+// Copyright (C) 2023 Anthony Frehner. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: pending
+description: Set.prototype.union should not call Set.prototype.add
+features: [Set-methods]
+---*/
+
+const s1 = new Set([1, 2]);
+const s2 = new Set([2, 3]);
+const expects = [1, 2, 3];
+
+const originalAdd = Set.prototype.add;
+let count = 0;
+Set.prototype.add = function (...rest) {
+  count++;
+  originalAdd.apply(this, rest);
+};
+
+const combined = s1.union(s2);
+
+combined.forEach(function (value) {
```

Comment:
This works but it is probably simpler to do include the `compareArray.js` harness file (by adding `includes: [compareArray.js]` to the frontmatter) and then replace this (and the `expects.length`/`combined.size` asserts) with

```js
assert.compareArray([...combined], expects);
```

Fine either way though.

Reply:
Updated all the tests! 👍